network/peer/peer.go

```diff
@@ -205,6 +205,7 @@ type peer struct {
 // Invariant: There must only be one peer running at a time with a reference to
 // the same [config.InboundMsgThrottler].
 func Start(
+	ctx context.Context,
```

Comment:
Since these functions are previously expected to run until there's an error or network shutdown adding a context to the signature means we need to be careful about what context gets passed in and whether cancelling it could cancel what are supposed to be long-running functions. Looking through the networking package for any case where this could break assumptions.